arch/src/aarch64/fdt.rs

```diff
@@ -484,7 +485,19 @@ fn create_devices_node<T: DeviceInfoForFDT + Clone + Debug>(
     fdt: &mut Vec<u8>,
     dev_info: &HashMap<(DeviceType, String), T>,
 ) -> Result<()> {
-    for ((device_type, _), info) in &*dev_info {
+    // We are trying to detect the root block device. We know
```

Comment:
This seems like a brute force approach. Is there any other simpler way to do this ?

Reply:
Ok, probably it wouldn't be worth it sending the id of the root block device as a parameter, but I would change the approach a little bit:

- iterate thorough the entire hashmap and remember the id with the lowest address. 
- get the device_info for the rootfs id and append it to the FDT first
- inside the for loop check if the current device type == TYPE_BLOCK && current id == rootfs_id . If it is, skip the entry

I don't think it would require more lines of code and it would help avoiding all the cloning.

Reply:
@serban300 thanks for the tips. They really helped. Take a look at how I changed the code. I got rid of all the clones but kept the sort since I though it was more elegant this way. Hope it is ok.

Reply:
Done, take a look.